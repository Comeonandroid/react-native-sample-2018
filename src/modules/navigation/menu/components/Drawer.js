
import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import CustomIcon from '../../../../utils/icon'
import { t } from '../../../../locales/i18n'

import { View, Image } from '../../../../elements'
import MenuItem from './MenuItem'

class DrawerContainer extends PureComponent {

  constructor(props) {
    super(props)

    this.navigate = this.navigate.bind(this)
    this.handleMenuClick = this.handleMenuClick.bind(this)
    this.handleGoHome = this.handleGoHome.bind(this)
    this.handleClose = this.handleClose.bind(this)
  }

  navigate(screen) {
    this.props.navigation.navigate(screen)
    this.props.navigation.navigate('DrawerClose')
  }

  handleGoHome(event) {
    console.log(event.target, 'dataset')
  }

  handleClose() {
    this.navigate('DrawerClose')
  }

  handleMenuClick(route) {
    this.navigate(route)
  }

  render() {
    return (
      <Overlay>
        <Header>
          <Logo source={require('../../../../../assets/icons/logo.png')}></Logo>
          <Habmburger onPress={this.handleClose}>
            <HamburgerIcon name="back" size={16} />
          </Habmburger>
        </Header>
        <View>
          <MenuItem
            icon={'trade'}
            route={'Trade'}
            onPress={this.handleMenuClick}
            size={14}
          >{t('menu.trade')}</MenuItem>
          <MenuItem
            text={t('menu.wallet')}
            icon={'wallet'}
            route={'Wallet'}
            onPress={this.handleMenuClick}
          >{t('menu.wallet')}</MenuItem>
          <MenuItem
            text={t('menu.settings')}
            icon={'settings'}
            route={'Settings'}
            onPress={this.handleMenuClick}
            size={18}
          >{t('menu.settings')}</MenuItem>
          <MenuItem
            text={t('menu.notifications')}
            icon={'info'}
            route={'Notifications'}
            onPress={this.handleMenuClick}
            size={18}
          >{t('menu.notifications')}</MenuItem>
          <MenuItem
            noBorder
            text={t('menu.contact')}
            icon={'contact'}
            route={'Contact'}
            onPress={this.handleMenuClick}
          >{t('menu.contact')}</MenuItem>
        </View>
      </Overlay>
    )
  }
}

const Overlay = styled(View)`
  background-color: ${ ({ theme }) => theme.colors.backgroundMenu};
  flex: 1;
  padding: 20px;
`

const Logo = styled(Image)`
  width: 36px;
  height: 58px;
  margin-left: 5px;
`

const Header = styled(View)`
  display: flex;
  align-items: center;
  flex-direction: row;
`
const Habmburger = styled.TouchableOpacity`
  display: flex;
  margin-left: auto;
`

const HamburgerIcon = styled(CustomIcon) `
  color: ${ ({ theme }) => theme.colors.drawerContrastColor}
`

DrawerContainer.propTypes = {
  navigation: PropTypes.object
}

export default DrawerContainer
