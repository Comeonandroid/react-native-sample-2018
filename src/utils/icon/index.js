import { createIconSetFromIcoMoon } from 'react-native-vector-icons'
import styled from 'styled-components'

import icoMoonConfig from './selection.json'
const CustomIcon = createIconSetFromIcoMoon(icoMoonConfig)

export default styled(CustomIcon) `
  color: ${ ({ theme, color }) => color || theme.colors.border };
`

// IMPORTANT
// If you want to add/remove icons please read this first
// https://www.reactnative.guide/12-svg-icons-using-react-native-vector-icons/12.1-creating-custom-iconset.html