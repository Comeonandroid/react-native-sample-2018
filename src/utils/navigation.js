import React from 'react'

import { t } from '../locales/i18n'

import HeaderMenuIcon from '../modules/navigation/header/components/HeaderMenuIcon'

export const getNavigationHeader = ({ navigation, screenProps }) => (
  {
    headerMode: 'float',
    title: t('route.' + navigation.state.routeName),
    headerTintColor: screenProps.theme.colors.headerContrastColor,
    headerStyle: {
      backgroundColor: screenProps.theme.colors.background,
      borderBottomColor: 'transparent',
    },
    headerRight: <HeaderMenuIcon navigation={navigation} />,
    // TODO: header title for all the pages
    headerBackTitle: null
  }
)

export const getHomeHeader = ({ navigation, screenProps }) => (
  {
    headerTintColor: screenProps.theme.colors.headerContrastColor,
    headerStyle: {
      backgroundColor: screenProps.theme.colors.background,
      borderBottomColor: 'transparent',
    },
    headerRight: <HeaderMenuIcon navigation={navigation} />
  }
)
