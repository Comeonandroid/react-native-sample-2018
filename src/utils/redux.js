import {
  createReactNavigationReduxMiddleware,
  createReduxBoundAddListener,
} from 'react-navigation-redux-helpers'

const reduxRouterMiddleware = createReactNavigationReduxMiddleware(
  "root",
  state => state.navigation,
)
const addListener = createReduxBoundAddListener('root')

export {
  reduxRouterMiddleware,
  addListener,
}
