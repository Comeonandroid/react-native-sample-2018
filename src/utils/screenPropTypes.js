import {
  PropTypes
} from 'prop-types'

export default {
  navigation: PropTypes.object.isRequired
}