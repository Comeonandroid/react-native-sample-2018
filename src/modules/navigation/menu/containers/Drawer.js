import { connect } from 'react-redux'
import { mapNavigatioToState } from '../../../../services/mapNavigationToState'

import Drawer from '../components/Drawer'

const mapStateToProps = ({ navigation }) =>
  mapNavigatioToState(navigation)

const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)(Drawer)
