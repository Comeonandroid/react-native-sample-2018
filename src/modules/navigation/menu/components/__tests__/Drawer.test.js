import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import { itBehavesLike } from '../../../../../specs/shared_examples/index'
import { wrapWithTheme } from '../../../../../specs/utils'

import Drawer from '../Drawer'

const setup = propOverrides => {
  const props = Object.assign({
    navigation: {
      navigate: jest.fn()
    },
  }, propOverrides)

  const component = <Drawer {...props} />
  const tree = renderer.create(wrapWithTheme(component))

  return {
    props,
    tree,
    component
  }
}

describe('<Drawer />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })
})
