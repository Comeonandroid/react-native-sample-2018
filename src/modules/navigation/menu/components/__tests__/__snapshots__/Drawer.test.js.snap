// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#282828",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
          undefined,
        ],
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/icons/logo.png",
        }
      }
      style={
        Array [
          Object {},
          Array [
            Object {
              "height": 58,
              "marginLeft": 5,
              "width": 36,
            },
            undefined,
          ],
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "marginLeft": "auto",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#fcfcfc",
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3c3c3c",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcfcfc",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#c1c1c1",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        TRADE
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 14,
            },
            Array [
              Object {
                "color": "#fcfcfc",
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3c3c3c",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcfcfc",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#c1c1c1",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        WALLET
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#fcfcfc",
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3c3c3c",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcfcfc",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#c1c1c1",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        SETTINGS
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            Array [
              Object {
                "color": "#fcfcfc",
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3c3c3c",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcfcfc",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#c1c1c1",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        NOTIFICATIONS
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            Array [
              Object {
                "color": "#fcfcfc",
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#3c3c3c",
          "borderBottomWidth": 0,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fcfcfc",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#c1c1c1",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        CONTACT US
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#fcfcfc",
              },
              Array [
                Object {
                  "color": "#c1c1c1",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`<Drawer /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "backgroundColor": "#5d5959",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          },
          undefined,
        ],
      ]
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../assets/icons/logo.png",
        }
      }
      style={
        Array [
          Object {},
          Array [
            Object {
              "height": 58,
              "marginLeft": 5,
              "width": 36,
            },
            undefined,
          ],
        ]
      }
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "display": "flex",
          "marginLeft": "auto",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#4b4c4d",
              },
              Array [
                Object {
                  "color": "#edecec",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#716d6d",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#edecec",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        TRADE
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 14,
            },
            Array [
              Object {
                "color": "#4b4c4d",
              },
              Array [
                Object {
                  "color": "rgb(255,255,255)",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#716d6d",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#edecec",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        WALLET
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#4b4c4d",
              },
              Array [
                Object {
                  "color": "rgb(255,255,255)",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#716d6d",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#edecec",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        SETTINGS
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            Array [
              Object {
                "color": "#4b4c4d",
              },
              Array [
                Object {
                  "color": "rgb(255,255,255)",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#716d6d",
          "borderBottomWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#edecec",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        NOTIFICATIONS
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 18,
            },
            Array [
              Object {
                "color": "#4b4c4d",
              },
              Array [
                Object {
                  "color": "rgb(255,255,255)",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#716d6d",
          "borderBottomWidth": 0,
          "display": "flex",
          "flexDirection": "row",
          "opacity": 1,
          "paddingBottom": 14,
          "paddingTop": 14,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Open Sans",
            },
            Array [
              Object {
                "color": "#edecec",
                "fontSize": 14,
                "fontWeight": "500",
                "letterSpacing": 0.6,
              },
              undefined,
            ],
          ]
        }
      >
        CONTACT US
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 16,
            },
            Array [
              Object {
                "color": "#4b4c4d",
              },
              Array [
                Object {
                  "color": "rgb(255,255,255)",
                  "marginLeft": "auto",
                },
                undefined,
              ],
            ],
            Object {
              "fontFamily": "icomoon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
